### ISSUE TITLE
formatRequestError

### ISSUE THREAD

**Issue by u9520107** [facebook/relay#325]
How do we deal with mutation failures?
More specifically, how do we return the cause of the failure such as 'not enough cash', or 'health too low'?
In my server side implementation, I am prompted to throw messages as errors in the mutateAndGetPayload function:
```
mutateAndGetPayload: () => {
...
  throw new Error('ran out of cache');
}
```

However, on the client-side:
```
Relay.Store.update(new ...Mutation(), {
  onFailure: (transaction) => {
    transaction.getError() // would throw
  }
});
```
The above code would throw error:
```
TypeError: Cannot read property 'map' of undefined
    at http://localhost:14403/jspm_packages/npm/react-relay@0.3.1/lib/RelayDefaultNetworkLayer.js:170:47
    at Array.map (native)
    at formatRequestErrors (http://localhost:14403/jspm_packages/npm/react-relay@0.3.1/lib/RelayDefaultNetworkLayer.js:164:17)
    at http://localhost:14403/jspm_packages/npm/react-relay@0.3.1/lib/RelayDefaultNetworkLayer.js:52:139
```
What formatRequestError was trying to do is the look for the locations property on the error object, which since it is not a syntaxError, locations is undefined.

How do we avoid this problem?

**Comment by josephsavona** [comment]
@u9520107 thanks for filing an issue - `RelayDefaultNetworkLayer` needs to be updated to handle non-syntax errors. 

For anyone interested in contributing, the fix should be a matter of testing for `locations` being null-ish, and just returning the `message` for that error.

**Comment by u9520107** [comment]
@josephsavona 
Ok. I'll try that out today and submit a PR later.

**Comment by steveluscher** [comment]
Closed by #334.
